src/libraries/Microsoft.NETCore.Platforms/tests/GenerateRuntimeGraphTests.cs

```diff
@@ -21,38 +23,219 @@ public GenerateRuntimeGraphTests(ITestOutputHelper output)
 
             Assert.NotEmpty(runtimeGroups);
 
+            return runtimeGroups;
+        }
+
+        private static ITaskItem CreateItem(ProjectItem projectItem)
+        {
+            TaskItem item = new TaskItem(projectItem.EvaluatedInclude);
+            foreach (var metadatum in projectItem.Metadata)
+            {
+                item.SetMetadata(metadatum.Name, metadatum.EvaluatedValue);
+            }
+            return item;
+        }
+
+        [Fact]
+        public void CanCreateRuntimeGraph()
+        {
             // will generate and compare to existing file.
             GenerateRuntimeGraph task = new GenerateRuntimeGraph()
             {
                 BuildEngine = _engine,
-                RuntimeGroups = runtimeGroups,
-                RuntimeJson = runtimeFile
+                RuntimeGroups = DefaultRuntimeGroupItems,
+                RuntimeJson = DefaultRuntimeFile,
+                UpdateRuntimeFiles = false
             };
             task.Execute();
 
             _log.AssertNoErrorsOrWarnings();
         }
 
-        private static ITaskItem CreateItem(ProjectItem projectItem)
+
+        [Fact]
+        public void CanIgnoreExistingInferRids()
         {
-            TaskItem item = new TaskItem(projectItem.EvaluatedInclude);
-            foreach(var metadatum in projectItem.Metadata)
+            // will generate and compare to existing file.
+            GenerateRuntimeGraph task = new GenerateRuntimeGraph()
             {
-                item.SetMetadata(metadatum.Name, metadatum.EvaluatedValue);
-            }
-            return item;
+                BuildEngine = _engine,
+                RuntimeGroups = DefaultRuntimeGroupItems,
+                RuntimeJson = DefaultRuntimeFile,
+                AdditionalRuntimeIdentifiers = new[] { "rhel.9-x64", "centos.9-arm64", "win-x64" },
+                UpdateRuntimeFiles = false
+            };
+
+            _log.Reset();
+            task.Execute();
+            _log.AssertNoErrorsOrWarnings();
+        }
+
+        /// <summary>
+        /// Runs GenerateRuntimeGraph task specifying AdditionalRuntimeIdentifiers then asserts that the
+        /// generated runtime.json has the expected additions (and no more).
+        /// </summary>
+        /// <param name="additionalRIDs">additional RIDs</param>
+        /// <param name="expectedAdditions">entries that are expected to be added to the RuntimeGraph</param>
+        /// <param name="additionalRIDParent">parent to use when adding a new RID</param>
+        /// <param name="runtimeFilePrefix">a unique prefix to use for the generated </param>
+        private void AssertRuntimeGraphAdditions(string[] additionalRIDs, RuntimeDescription[] expectedAdditions, string additionalRIDParent = null, [CallerMemberName] string runtimeFilePrefix = null)
+        {
+            string runtimeFile = runtimeFilePrefix + ".runtime.json";
```

Comment:
Should these files be created on a temp directory and cleaned up after the test class is disposed? 

We have: https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs to achieve that.

Reply:
Test working directory is a temp directory. I don’t think cleaning them up is desirable, they are useful for debugging.

Reply:
Ok, as long as the working dir is a temp directory, I'm fine with it not being cleaned up.

Reply:
I'm calling it temp in the context of it's the test output directory.  It will be cleaned up by normal dev activities.  I intentionally don't want to hide these files in some random location as they are useful for diffing what the result of the test is.